torchgeo/datasets/cropharvest.py

```diff
@@ -0,0 +1,323 @@
+
+        Raises:
+            DatasetNotFoundError: If dataset is not found and *download* is False.
+            ImportError: If h5py is not installed
+        """
+        try:
+            import h5py  # noqa: F401
+        except ImportError:
+            raise ImportError(
+                "h5py is not installed and is required to use this dataset"
+            )
+
+        self.root = root
+        self.transforms = transforms
+        self.checksum = checksum
+        self.download = download
+
+        self._verify()
+
+        self.files = self._load_features(self.root)
+        self.labels = self._load_labels(self.root)
+        self.classes = self.labels["properties.label"].unique()
+        self.classes = np.insert(self.classes, 0, ["None", "Other"])
+
+    def __getitem__(self, index: int) -> dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            single pixel timeseries array and label at that index
+        """
+        files = self.files[index]
+        data = self._load_array(files["chip"])
+
+        label = self._load_label(files["index"], files["dataset"])
+        sample = {"array": data, "label": label}
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.files)
+
+    def _load_features(self, root: str) -> list[dict[str, str]]:
+        """Return the paths of the files in the dataset.
+
+        Args:
+            root: root dir of dataset
+
+        Returns:
+            list of dicts containing path for each of hd5 single pixel time series and
+            its key for associated data
+        """
+        files = []
+        chips = glob.glob(
+            os.path.join(root, self.file_dict["features"]["extracted_filename"], "*.h5")
+        )
+        chips = sorted(os.path.basename(chip) for chip in chips)
+        for chip in chips:
+            chip_path = os.path.join(
+                root, self.file_dict["features"]["extracted_filename"], chip
+            )
+            index = chip.split("_")[0]
+            dataset = chip.split("_")[1][:-3]
+            files.append(dict(chip=chip_path, index=index, dataset=dataset))
+        return files
+
+    def _load_labels(self, root: str) -> pd.DataFrame:
+        """Return the paths of the files in the dataset.
+
+        Args:
+            root: root dir of dataset
+
+        Returns:
+            pandas dataframe containing label data for each feature
+        """
+        filename = self.file_dict["labels"]["extracted_filename"]
+        with open(os.path.join(root, filename), encoding="utf8") as f:
+            data = json.load(f)
+            df = pd.json_normalize(data["features"])
```

Comment:
Remind me why this is needed? Can't we use `json.load` only? If not, can we instead use `pd.read_json`?

Reply:
From my testing, pd.read_json fails on the data because the list of objects to make the dataframe is not at the root. So the data must first be loaded and then the dataframe created with ["features"] as the root. The json_normalize call flattens the nested objects into the pandas dataframe to make queries and calls to .unique() possible rather than working with the list directly. Is there an alternative way you would suggest going about this?

Reply:
Nah, that seems reasonable to me. Thanks for clarifying!